### PR TITLE
boards: mbed_lpc1768: Improve documentation

### PR DESCRIPTION
**Pull request by francois-berder**
### Contribution description

This commit improves the documentation for the mbed lpc1768 board.

### Testing procedure

None

### Issues/PRs references

None

### CODE REVIEW

File: boards/mbed_lpc1768/doc.txt [hunk 1]
```diff
@@ -3,24 +3,95 @@
...
+[GNU Tools for ARM Embedded Processors](https://launchpad.net/gcc-arm-embedded)
+toolchain.
+
+## Flashing the device
+
+The easiest way to flash the device is to connect to the board via USB. The
+board appears as a 2MB usb drive. Simply copy your binary file to the root
+directory and reset your board.
+While the LPC1768 is flashed by the MBED-IF01 with the new firmware, the
+status LED blinks fast.
```
Comment (benpicco) [inline boards/mbed_lpc1768/doc.txt]: I suppose `make flash` still works? (IMHO that's much easier workflow-wise than copying files)

Comment (francois-berder) [reply boards/mbed_lpc1768/doc.txt]: Yes `make flash` still works and its sole purpose is to copy the binary file to the USB flash drive. I also mentioned that the user needs to press the reset button to start the new firmware.

**Comment by benpicco** [comment]
I wonder how similar the missing peripherals are to lpc23xx

**Comment by francois-berder** [comment]
Sorry @benpicco there was a typo in the doxygen header so I fixed it as part of this commit.

**Pull request merged**